### DISCUSSION TITLE
Add border to BSP layout when only 1 window is open - Possibility to make bar rounded

### DISCUSSION THREAD

**Discussion by emanuelep57** [qtile/qtile#2985] [Ideas]
Hello guys, I hope I am not the only one loving the layout of bspwm and its style but everything else of qtile. However when just 1 window is open in the workspace it doesn't have any border, which makes it kind of not nice to look at.

Also it would be nice to make the bar rounded without having to use picom, because in case we want to use picom then also all our windows will have rounded borders unless we exclude them singularly

**Comment by m-col** [comment]
> Hello guys, I hope I am not the only one loving the layout of bspwm and its style but everything else of qtile. However when just 1 window is open in the workspace it doesn't have any border, which makes it kind of not nice to look at.

Many layouts have `border_on_single` (and `margin_on_single`) boolean options to configure this. In fact this was implemented today for the `Tile` layout: https://github.com/qtile/qtile/pull/3023. A PR to add this to the Bsp layout would be welcome! :)

> Also it would be nice to make the bar rounded without having to use picom, because in case we want to use picom then also all our windows will have rounded borders unless we exclude them singularly

I do not expect we will support rounded bar corners internally but you could white-list the picom rule to the bar rather than black-list others. The bar window has a `QTILE_INTERNAL` property with value `1` which can be used as a picom rule to identify the window.
